### ISSUE TITLE
Markdown: Long headings broken (also unstable)

### ISSUE THREAD

**Issue by SimenB** [prettier/prettier#3169]
**Prettier 1.8.0**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBidACABgJzgBwh3hwGcsMBtAQxx2oE8AeAWwgBMBXAGzgDlqLOBgA+VNl14ChAGgwR8MAJbRSAXQB8agDpRd6VNjyFicMhRp1GrDj36DhYyhLvS4M3QuWrNOqCBkQLxUoUmRQalIYZAAzam5SdxAAI3owAGs4GABlfGowJSgAc2QYHE4k9ggwWPjEwIArUgAPACE0zJyHABlCuFqEpIhOGHwRgCYB+pA8sjNkEBZadKqAd39A-BxCmAB1JXYYAAtkAA4ABkDEliVS8qTSQqLeAEVOCHgppJhqZP3Dk5IcaBMrUJTcJ4AYQgLCWCyg0H6gU4iQAKr8wkg4oMAL44oA)
```sh
--parser markdown
```

**Input:**
```markdown
### `reporters` [array<moduleName | [moduleName, options]>]

### `reporters` [array<moduleName | [moduleName,
options]>]

```

**Output:**
```markdown
### `reporters` [array<moduleName | [moduleName,
options]>]

### `reporters` [array<moduleName | [moduleName,

options]>]

```

**Expected behavior:**
No change on input

**Comment by SimenB** [comment]
This is extra weird as the heading has a width of 60, so it shouldn't be broken anyways

**Comment by azz** [comment]
This has been fixed in `1.8.1`.
